app/code/core/Mage/Adminhtml/Block/Notification/Toolbar.php

```diff
@@ -53,13 +53,8 @@ public function isShow()
         if ($this->getRequest()->getControllerName() === 'notification') {
             return false;
         }
-        if ($this->getCriticalCount() == 0 && $this->getMajorCount() == 0 && $this->getMinorCount() == 0
-            && $this->getNoticeCount() == 0
-        ) {
-            return false;
-        }
-
-        return true;
+        return !($this->getCriticalCount() == 0 && $this->getMajorCount() == 0 && $this->getMinorCount() == 0
+            && $this->getNoticeCount() == 0);
```

Comment:
We can simplify with DeMorgan's Law `!(A && B) == (!A || !B)`:
```suggestion
        return $this->getCriticalCount() || $this->getMajorCount() || $this->getMinorCount() || $this->getNoticeCount();
```

Reply:
Step by step ... :)

https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#simplifydemorganbinaryrector

Reply:
Oh...so rector rules cannot be chained?

Reply:
With rulessets (chained rules) the PRs would grow to large. Mhhh.

Reply:
Mhhh .... https://getrector.com/demo/ec03b365-db13-4e75-8ecd-d3773ae96870

Reply:
I created an issue: https://github.com/rectorphp/rector/issues/8781